yarl/_url.py

```diff
@@ -1062,24 +1064,45 @@ def join(self, url):
 
     def human_repr(self):
         """Return decoded human readable string for URL representation."""
-
+        user = _human_quote(self.user, "#/:?@")
+        password = _human_quote(self.password, "#/:?@")
+        host = self.host
+        if host:
+            host = self._encode_host(self.host, human=True)
+        path = _human_quote(self.path, "#?")
+        query_string = "&".join(
+            f'{_human_quote(k, "#&+;=")}={_human_quote(v, "#&+;=")}'
+            for k, v in self.query.items()
```

Comment:
I have a hunch that this may break backward compatibility with the old `;` subdelimiter (https://github.com/cherrypy/cherrypy/issues/1860#issuecomment-640246780). Could you please ensure that it's explicitly covered with tests?

Reply:
`;` is a delimiter between name-value pairs (as well as `&`). It is replaced with `&` when convert URL to string (either by `str()` or by `URL.human_repr()`). `;` in the name or value will be %-encoded in output, so they will not be delimiters.

There is a test for all printable non-alphanumeric ASCII characters (`test_human_repr_delimiters`) which covers this case.